tasks/mcp.js

```diff
@@ -0,0 +1,103 @@
+const { task } = require('hardhat/config')
+const axios = require('axios')
+const fs = require('fs').promises
+const FormData = require('form-data')
+require('dotenv').config()
+
+task('mcp', 'Upload file to MCP and mint NFT')
+  .addParam('file', '')
+  .addOptionalParam('name', '')
+  .addOptionalParam('desc', '')
+  .setAction(async (taskArgs) => {
+    if (!network.config.mcp.authToken) {
+      console.log('Please Login for Auth Token')
+    } else {
+      const file = await fs.readFile(taskArgs.file) // read file
+
+      console.log('Uploading file to MCP...')
+      const upload = await mcpUpload(taskArgs.file.split('/').pop(), file) // upload file to MCP
+
+      // generate metadata using the file CID
+      // currently MCP API does not show txHash
+      console.log('Generating metadata...')
+      const metadata = generateMetadata(
+        taskArgs.name || `${taskArgs.file.split('.')[0]}.json`,
+        taskArgs.desc,
+        upload.cid,
+        //upload.txHash,
+      )
+
+      // Display JSON file
+      console.log(JSON.stringify(metadata))
+
+      // upload JSON to MCP
+      console.log('Uploading metadata to MCP...')
+      const metadataUpload = await mcpUpload(
+        metadata.name,
+        JSON.stringify(metadata),
+      )
+
+      await mint(metadataUpload.cid) // mint NFT!
+    }
+  })
+
+const mcpUpload = async (fileName, file, duration = 180) => {
+  const form = new FormData()
+  form.append('duration', duration)
+  form.append('file', file, fileName)
+
+  let result = { cid: '' }
+
+  await axios
```

Comment:
you can take a look on how to use async/await for axios
it's like you combine async/await with returning Promise